### PR TITLE
Fix validation in kubectl create ingress

### LINKED ISSUE #1041: kubectl create ingress panics when wrong annotation is passed
**What happened**:

```golang
kubectl create ingress lala --class=default --rule="foo.com/bar=svc1:8080,tls=my-cert" --annotation=lalala
panic: runtime error: index out of range [1] with length 1
```

**What you expected to happen**:
No panic to occur
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
I need to parse my arrays better :P

/kind bug
/sig cli
/triage accepted

### PR DESCRIPTION
**Pull request by rikatz**
Signed-off-by: Ricardo Pchevuzinske Katz <ricardo.katz@gmail.com>

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
When running `kubectl create ingress name --annotation=key` without '=value' it panics due to lack of validation

#### Which issue(s) this PR fixes:
Fixes https://github.com/kubernetes/kubectl/issues/1041

```release-note
Fix panic with kubectl create ingress annotation flag and empty value
```
This was added in PR #94327

/sig cli
/priority important-soon

### CODE REVIEW

**Review by eddiezane — APPROVED** [review]
Good catch and thank you for adding a test for this 😄 .

This probably needs a release note?

**Comment by rikatz** [comment]
Added the release note, will cherry pick to v1.20 and v1.21 once merged :D

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go [hunk 1]
```diff
@@ -160,7 +160,7 @@ func NewCmdCreateIngress(f cmdutil.Factory, ioStreams genericclioptions.IOStream
 	cmd.Flags().StringVar(&o.IngressClass, "class", o.IngressClass, "Ingress Class to be used")
 	cmd.Flags().StringArrayVar(&o.Rules, "rule", o.Rules, "Rule in format host/path=service:port[,tls=secretname]. Paths containing the leading character '*' are considered pathType=Prefix. tls argument is optional.")
 	cmd.Flags().StringVar(&o.DefaultBackend, "default-backend", o.DefaultBackend, "Default service for backend, in format of svcname:port")
-	cmd.Flags().StringArrayVar(&o.Annotations, "annotation", o.Annotations, "Annotation to insert in the ingress object, in the format annotation=value")
+	cmd.Flags().StringArrayVar(&o.Annotations, "annotation", o.Annotations, "Annotation to insert in the ingress object, in the format annotation=[value]")
```
Comment (KnVerey) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go:163]: Nit: Are the square brackets to denote that technically the value can be empty/omitted? I feel like that use case might be rare enough for the brackets to cause more confusion than clarification.

Comment (rikatz) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: fixed, thank you!!

**Pull request merged**